examples/__tests__/cognito.test.ts

```diff
@@ -0,0 +1,503 @@
+            username: randomEmail,
+            userPoolId,
+          }),
+        ).toExistAndMatchObject({
+          Username: randomEmail,
+          Enabled: true,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: existingUser,
+            },
+          ]),
+        });
+      } catch (e) {
+        expect(e).toMatchInlineSnapshot(
+          `[Error: Expected "us-east-1_2KFDP3x2n" user pool to have a user with username "ci@nen.ma"]`,
+        );
+      }
+
+      try {
+        await expect(
+          cognitoUser({
+            username: existingUser,
+            userPoolId,
+            clientConfig: {
+              region: 'us-east-2',
+            },
+          }),
+        ).toExistAndMatchObject({
+          id: '456',
+          title: 'Play Fifa',
+        });
+      } catch (e) {
+        expect(e).toMatchInlineSnapshot(
+          `[ResourceNotFoundException: User pool us-east-1_2KFDP3x2n does not exist.]`,
+        );
+      }
+    });
+  });
+
+  describe('.not.toExistAndMatchObject', () => {
+    it('should fail when user exists and matches the expected object', async () => {
+      try {
+        await expect(
+          cognitoUser({
+            username: existingUser,
+            userPoolId,
+            retryPolicy: {
+              retries: 0,
+            },
+          }),
+        ).not.toExistAndMatchObject({
+          Username: existingUser,
+          Enabled: true,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: existingUser,
+            },
+          ]),
+        });
+      } catch (e) {
+        expect(e).toHaveProperty('message');
+        expect(e).toHaveProperty('matcherResult.message');
+        expect(e).toHaveProperty('matcherResult.actual');
+        expect(e).toHaveProperty('matcherResult.expected', {
+          Username: existingUser,
+          Enabled: true,
+          UserAttributes: expect.arrayContaining([
+            {
+              Name: 'email',
+              Value: existingUser,
+            },
+          ]),
+        });
+      }
+    });
+
+    it("should succeed when user exists but doesn't match the expected object", async () => {
+      await expect(
+        cognitoUser({
+          username: existingUser,
+          userPoolId,
+        }),
+      ).not.toExistAndMatchObject({
+        Username: existingUser,
+        Enabled: false,
+      });
+    });
+
+    it('should succeed when user does not exists', async () => {
+      const randomEmail = chance.email();
+      await expect(
+        cognitoUser({
+          username: randomEmail,
+          userPoolId,
+        }),
+      ).not.toExistAndMatchObject({
+        Username: randomEmail,
+        Enabled: true,
+      });
+    });
+  });
+
+  describe('toExistAndMatchSnapshot', () => {
+    // TODO
+    it.skip('should print values when user exists', async () => {
```

Comment:
what is missing?

Reply:
As far as I remember generated snapshots had dynamic values so this is why I (temporarily) skipped this test causing test failures, we probably need to use `expect.any(String)` to replace dynamic values... I'll look into it.

Thank you for bringing this to my attention, and I apologize for the delayed response.

Reply:
Fixed in this commit https://github.com/serverless-guru/sls-jest/pull/73/commits/69068078bcbe1308761d7562e1772d703598537c